### PR TITLE
Fixed #23926 -- Show better validation error when creating custom permis...

### PR DESCRIPTION
**Pull request by joeribekker**
...sions that are too long.

### CODE REVIEW

**Comment by erikr** [comment]
buildbot, test this please

File: tests/auth_tests/test_management.py [hunk 1]
```diff
@@ -565,3 +565,16 @@ def test_verbose_name_length(self):
...
+    def test_custom_permission_name_length(self):
+        auth_app_config = apps.get_app_config('auth')
+
+        ContentType.objects.get_by_natural_key('auth', 'permission')
+        custom_perm_name = 'a' * 256
+        models.Permission._meta.permissions = [
+            ('my_custom_permission', custom_perm_name),
+        ]
+
+        six.assertRaisesRegex(self, exceptions.ValidationError,
```
Comment (timgraham) [inline tests/auth_tests/test_management.py]: prefer the context manager version of `self.assertRaisesMessage()`

File: tests/auth_tests/test_management.py [hunk 2]
```diff
@@ -565,3 +565,16 @@ def test_verbose_name_length(self):
         six.assertRaisesRegex(self, exceptions.ValidationError,
             "The verbose_name of auth.permission is longer than 244 characters",
             create_permissions, auth_app_config, verbosity=0)
+
+    def test_custom_permission_name_length(self):
+        auth_app_config = apps.get_app_config('auth')
+
+        ContentType.objects.get_by_natural_key('auth', 'permission')
+        custom_perm_name = 'a' * 256
+        models.Permission._meta.permissions = [
```
Comment (timgraham) [inline tests/auth_tests/test_management.py]: need to restore the original value of `models.Permission._meta.permissions` in a try/finally block so the test doesn't leak state?

Comment (joeribekker) [inline tests/auth_tests/test_management.py]: You're right. Although I see more tests that leak some state, it's no reason to have new code do the same. I'll update the PR.

**Comment by timgraham** [comment]
merged in 0ed20d5cc4bf94646ffd4e4fcbd963b9916039cd, thanks!

**Pull request closed without merging**